### ISSUE TITLE
ResourceReference chooser only supports Maven based plugins

### ISSUE THREAD

**Issue by greenwoodma** [GateNLP/gate-core#33]
Currently (and I can understand why) the new ResourceReference chooser only supports viewing resources inside plugins which are instances of Plugin.Maven. While I think that currently covers all cases, long term we should probably change this as there is nothing stopping over types of plugins offering resources. I guess we should show any plugin that returns true for `hasResources` although we might need to add a new method that can enumerate the resources without copying them to Plugin to support this.

**Comment by ianroberts** [comment]
> we might need to add a new method that can enumerate the resources without copying them to Plugin to support this.

I added such a method to `Plugin.Maven` specifically but it could be good to generalise this to all `Plugin` types.

**Comment by greenwoodma** [comment]
Yes, I think it would be good if we did (although not urgent). I guess there are other issues though around distinguishing between plugins with the same name, where the maven specific details are also used. Not sure how general we can make it, or how much it matters given the only plugin types where this currently makes much difference is the maven based ones (although it might be nice to include directory plugins in the new tab as well even if they return a file:// as it cuts down the search space)
